File: ios/ColocKitchenRace/Models/DietaryPreference.swift
//
//  DietaryPreference.swift
//  ColocKitchenRace
//
//  Created by Julien Rahier on 07/02/2026.
//

import Foundation

enum DietaryPreference: String, Codable, CaseIterable, Identifiable, Hashable {
    case vegetarian = "vegetarian"
    case vegan = "vegan"
    case glutenFree = "gluten_free"
    case lactoseFree = "lactose_free"
    case nutFree = "nut_free"

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .vegetarian: return String(localized: "Vegetarian")
        case .vegan: return String(localized: "Vegan")
        case .glutenFree: return String(localized: "Gluten free")
        case .lactoseFree: return String(localized: "Lactose free")
        case .nutFree: return String(localized: "Nut free")
        }
    }

    var icon: String {
        switch self {
        case .vegetarian: return "🥬"
        case .vegan: return "🌱"
        case .glutenFree: return "🌾"
        case .lactoseFree: return "🥛"
        case .nutFree: return "🥜"
        }
    }
}

File: android/app/src/main/java/dev/rahier/colocskitchenrace/data/model/DietaryPreference.kt
package dev.rahier.colocskitchenrace.data.model

enum class DietaryPreference(val displayName: String, val icon: String) {
    VEGETARIAN("Vegetarien", "\uD83E\uDD66"),
    VEGAN("Vegan", "\uD83C\uDF31"),
    GLUTEN_FREE("Sans gluten", "\uD83C\uDF3E"),
    LACTOSE_FREE("Sans lactose", "\uD83E\uDD5B"),
    NUT_FREE("Sans noix", "\uD83E\uDD5C");

    companion object {
        fun fromFirestore(value: String): DietaryPreference? = when (value) {
            "vegetarian" -> VEGETARIAN
            "vegan" -> VEGAN
            "gluten_free" -> GLUTEN_FREE
            "lactose_free" -> LACTOSE_FREE
            "nut_free" -> NUT_FREE
            else -> null
        }
    }

    fun toFirestore(): String = when (this) {
        VEGETARIAN -> "vegetarian"
        VEGAN -> "vegan"
        GLUTEN_FREE -> "gluten_free"
        LACTOSE_FREE -> "lactose_free"
        NUT_FREE -> "nut_free"
    }
}
